messaging/src/test/java/org/axonframework/eventhandling/tokenstore/jpa/JpaTokenStoreTest.java

```diff
@@ -281,17 +274,60 @@ public void testQuerySegments() {
             final int[] segments = jpaTokenStore.fetchSegments("proc2");
             assertThat(segments.length, is(1));
         }
-
         {
             final int[] segments = jpaTokenStore.fetchSegments("proc3");
             assertThat(segments.length, is(0));
         }
 
+        entityManager.flush();
+        entityManager.clear();
+    }
+
+    @Transactional
+    @Test
+    public void testQueryAvailableSegments() {
+        prepareTokenStore();
+
+        {
```

Comment:
I am curious about these blocks around the test units... Not sure why those are there. So, fair of you to copy them, but I think you don't necessarily have to.

Reply:
The blocks ensure we can reuse the names of final variables, instead of using `segments1`, `segments2`, etc. I don't really have any preference here, as long as we're consistent, As such, I had decided to follow the same patterns as in some other tests.

Reply:
Ah, of course, glanced over that, I guess. I agree, though, that the consistency here is helpful to understand the tests better. So, let's leave it as you've set it up right now.